packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -396,25 +396,16 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 					'error',
 					errorListener
 				);
-
-				try {
-					resolve(
-						/**
-						 * This is awkward, but Asyncify makes wasm_sapi_handle_request return
-						 * Promise<Promise<number>>.
-						 *
-						 * @TODO: Determine whether this is a bug in emscripten or in our code.
-						 */
-						await await this[__private__dont__use].ccall(
-							'wasm_sapi_handle_request',
-							NUMBER,
-							[],
-							[]
-						)
-					);
-				} catch (e) {
-					reject(e);
+				const response = this[__private__dont__use].ccall(
+					'wasm_sapi_handle_request',
+					NUMBER,
+					[],
+					[]
+				);
+				if (response instanceof Promise) {
+					return response.then(resolve).catch(reject);
```

Comment:
This will reject on errors in `response` and not in `resolve`. I _think_ the latter will be handled upstream:

```suggestion
					return response.then(resolve, reject);
```

I'm not 100% confident here, what do you think?

Reply:
I think you are right on that!
In the meantime, I found out why we get:
```
(node:399877) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unhandledRejection listeners added to [process]. Use emitter.setMaxListeners() to increase limit
```
We reject the promise, and then the errorListener is called and reject is called again.
This cause the above error.